src/scraping/tools.py

```diff
@@ -182,23 +186,21 @@ class DataBaseInteraction:
     table_name (str): name of the table to be created/used in the database
     """
 
-    def __init__(
-        self,
-        dataframe: pd.DataFrame,
-        table_name: str
-                ) -> None:
-
+    def __init__(self) -> None:
```

Comment:
re-add data frame and table_name, and set dataframe to  = None so it is not mandatory to pass when instantiating the object